### ISSUE TITLE
[BUG] Uploading a directory doesn't work

### ISSUE THREAD

**Issue by D3X** [jfurrow/flood#465]
As noted in #451, uploading a directory (either via the "add torrent" dialog, or by drag-n-dropping onto the torrent list) results in an error that is not even handled by the UI.

Proposed solution:
Utilize the [javascript API](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryEntry/createReader) to read the directory and add all the files to the list.
Handle failures in a nice manner, probably displaying a message similar to the "success" one.

Questions:
1. Should we traverse the directory tree? I can't see a use for that, but maybe some people would like it. In theory it's easy to implement, but we should be careful with circular symlinks.
2. Do we want to filter files out and only try to upload the .torrent ones? Seems like a good idea to me, but maybe there's a use case I didn't think of.
3. Limit the number of files to collect? What if someone drops a folder with a million .torrent files? Do we push them to the server and blame the user if something breaks? I guess the upload would timeout or exceed the maximum post size anyway...
4. In the "add torrent" dialog, should the files be added to the list, or do we want to stick to the directory itself, the way it currently works? I think it would be easier to leave it the way it is and handle the reading/traversal in the uploader itself.
5. What happens to empty directories? Display an error message upon reading the directory?

@jfurrow once we work out answers for those questions, I can work on a fix. It's definitely not a high priority bug, though.

**Comment by jfurrow** [comment]
These are good questions...

After reading this, I advise just disabling uploading directories entirely. Let's handle it gracefully in the UI, but let's not allow anyone to specify a directory when uploading files.

If we can detect it's a directory while the user is dragging onto the `Dropzone` component, let's display that in the component itself (maybe with a big "x" icon and some explanatory text). The same logic could apply to anything that's not a `.torrent` file, but that can be built separately.

@D3X What do you think?

**Comment by D3X** [comment]
@jfurrow yeah, that makes sense. I doubt there's much demand for uploading whole directories.

We can detect directories in the `dropend` callback or anywhere where we have access to the `DataTransfer` object. Not sure about "while dragging".
